source/extensions/router/cluster_specifiers/lua/lua_cluster_specifier.cc

```diff
@@ -0,0 +1,91 @@
+#include "source/extensions/router/cluster_specifiers/lua/lua_cluster_specifier.h"
+
+#include "source/common/router/config_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Router {
+namespace Lua {
+
+PerLuaCodeSetup::PerLuaCodeSetup(const std::string& lua_code, ThreadLocal::SlotAllocator& tls)
+    : lua_state_(lua_code, tls) {
+  lua_state_.registerType<HeaderMapWrapper>();
+  lua_state_.registerType<RouteHandleWrapper>();
+
+  const Filters::Common::Lua::InitializerList initializers;
+
+  cluster_function_slot_ = lua_state_.registerGlobal("envoy_on_route", initializers);
+  if (lua_state_.getGlobalRef(cluster_function_slot_) == LUA_REFNIL) {
+    ENVOY_LOG(info, "envoy_on_route() function not found. Lua will not hook cluster specifier.");
```

Comment:
I think the configuration should be rejected by an exception if there is no envoy_on_route method.

Reply:
done, thanks.